samples/wifi/scan/src/main.c

```diff
@@ -221,44 +244,44 @@ static int wifi_scan(void)
 {
 	struct net_if *iface = net_if_get_default();
 
-	for (int i = 0; i < ARRAY_SIZE(tests); i++) {
-		struct wifi_scan_params params = tests[i];
-#ifdef CONFIG_WIFI_SCAN_SSID_FILT_MAX
-		if (!strlen(params.ssids[0])) {
-			char *buf = NULL;
-
-			buf = malloc(strlen(CONFIG_WIFI_SCAN_SSID_FILT) + 1);
-			if (!buf) {
-				LOG_ERR("Malloc Failed");
-				return -EINVAL;
-			}
-
-			strcpy(buf, CONFIG_WIFI_SCAN_SSID_FILT);
-			if (wifi_utils_parse_scan_ssids(buf,
-							params.ssids)) {
-				LOG_ERR("Incorrect value(s) in CONFIG_WIFI_SCAN_SSID_FILT: %s",
-						CONFIG_WIFI_SCAN_SSID_FILT);
-				return -EINVAL;
-			}
+	struct wifi_scan_params params = tests[0];
+
+	if (strlen(CONFIG_WIFI_SCAN_BANDS_LIST)) {
+		char *buf = malloc(strlen(CONFIG_WIFI_SCAN_BANDS_LIST) + 1);
```

Comment:
Can sizeof be used instead of `strlen`, as far as I can tell the `CONFIG_WIFI_SCAN_BANDS_LIST` evaluates to literal of size known at compile time.

Reply:
Size of in line 249?

Reply:
@de-nordic 
when we use sizeof on "" (default value of CONFIG_WIFI_SCAN_BANDS_LIST), it returns 1 (for the null terminator), and the if() condition in the above code passes all the time which is not expected.
Using sizeof, we may not achieve desired result in this specific case, so using strlen to check for an empty string is more appropriate.